.github/workflows/ci.yml

```diff
@@ -140,3 +140,31 @@ jobs:
       with:
         token: ${{ secrets.GITHUB_TOKEN }}
         args: --workspace --all-features --all-targets -- -D warnings
+  minimal_versions:
+    name: Minimal versions
+    needs: smoke
+    strategy:
+      matrix:
+        os: [ "ubuntu-latest", "windows-latest", "macos-latest" ]
+        rust: [ "1.54.0" ] # MSRV
+    continue-on-error: ${{ matrix.rust != 'stable' }}
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+         profile: minimal
+         toolchain: nightly
+      - uses: actions-rs/toolchain@v1
+        with:
+         profile: minimal
+         toolchain: ${{ matrix.rust }}
+         override: true
+      - name: Downgrade crates to minimal supported versions
+        run: cargo +nightly update -Z minimal-versions
+      - uses: Swatinem/rust-cache@v1
+      - name: Default features
+        run: cargo test --workspace
+      - name: No-default features
+        run: cargo test --workspace --no-default-features
```

Comment:
I think I'd rather these be `check` than `test`

Like with MSRV, I'm assuming 99% of the time, a minimal version issue arises from an incompatibility with the API rather than a behavior change.

You have more experience with this though, so let me know if you've been seeing significant issues with relying on new behavior with the same APIs

Reply:
Another benefit to only doing `check` (rather than `test` or `check --all-targets`) is it means we don't have to block this on dev dependencies (granted, I maintain the only dev-dependency in this crate)

Reply:
@epage 

> Like with MSRV, I'm assuming 99% of the time, a minimal version issue arises from an incompatibility with the API rather than a behavior change.

Yeah, but there is still that 1% when your minimal crate version is some bugfix, that is tested inside the crate.

> You have more experience with this though, so let me know if you've been seeing significant issues with relying on new behavior with the same APIs

In my experience this helped 1 or 2 times. 
But I think, that if we can automate something to avoid potential problems, we should do this all in. Tradeoff here is maybe somewhat slower CI, but all jobs run in parallel and additional wait time is just waiting in the queue for jobs to start.

Reply:
> But I think, that if we can automate something to avoid potential problems, we should do this all in. Tradeoff here is maybe somewhat slower CI, but all jobs run in parallel and additional wait time is just waiting in the queue for jobs to start.

So the impact here is less but in clap, we were doing the "all in" approach before and the CI took hours to run.  Some jobs alone took 30+ minutes to run.  Another problem is that parallelism is limited; [Github limits the number of concurrent runners](https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration) and I believe its global to the org the repo is in.  I also feel we should be considerate of the resources Github is giving to us for free.

We then switched to a two-tier CI system where we checked fewer things on to give a green light for the PR but then bors would run the full set on merge.  This still took somewhere between 30-60 minutes to run.  We eventually dropped the "all in" approach and CI gives us nearly the same coverage for a fraction of the time, focusing on the likely problems